implant/sliver/handlers/handlers_linux.go

```diff
@@ -60,10 +72,182 @@ var (
+
+	chmod.Response = &commonpb.Response{}
+	if err == nil {
+		// Convert string to octal number
+		octal, err := strconv.ParseInt(chmodReq.FileMode, 8, 32)
+		if err == nil {
+
+			setuid := octal & 04000
+			setgid := octal & 02000
+			setstcky := octal & 01000
+
+			// Cast the octal number to fs.FileMode
+			fileMode := os.FileMode(octal)
+
+			// Found this was necessary because the constructor above doesn't set special permissions
+			if setuid > 0 {
+				fileMode = fileMode | os.ModeSetuid
+			}
+			if setgid > 0 {
+				fileMode = fileMode | os.ModeSetgid
+			}
+			if setstcky > 0 {
+				fileMode = fileMode | os.ModeSticky
+			}
+
+			if chmodReq.Recursive {
+
+				err := filepath.WalkDir(target, func(file string, d fs.DirEntry, err error) error {
+					if err == nil {
+						err = os.Chmod(file, fileMode)
+						if err != nil {
+							return err
+						}
+					} else {
+						return err
+					}
+					return nil
+				})
+				if err != nil {
+					chmod.Response.Err = err.Error()
+				}
+
+			} else {
+				err = os.Chmod(target, fileMode)
+				if err != nil {
+					chmod.Response.Err = err.Error()
+				}
+			}
+		} else {
+			chmod.Response.Err = err.Error()
+		}
+	} else {
+		chmod.Response.Err = err.Error()
+	}
+
+	data, err = proto.Marshal(chmod)
+	resp(data, err)
+}
+
+func chownHandler(data []byte, resp RPCResponse) {
+	chownReq := &sliverpb.ChownReq{}
+	err := proto.Unmarshal(data, chownReq)
+	if err != nil {
+		// {{if .Config.Debug}}
+		log.Printf("error decoding message: %v", err)
+		// {{end}}
+		return
+	}
+
+	chown := &sliverpb.Chown{}
+	target, _ := filepath.Abs(chownReq.Path)
+	chown.Path = target
+	_, err = os.Stat(target)
+
+	chown.Response = &commonpb.Response{}
+	if err == nil {
+		uid_str := chownReq.Uid
+		usr, err := user.Lookup(uid_str)
+		if err == nil {
+			uid, err := strconv.ParseInt(usr.Uid, 10, 32)
+			if err == nil {
+				gid_str := chownReq.Gid
+				grp, err := user.LookupGroup(gid_str)
+				if err == nil {
+					gid, err := strconv.ParseUint(grp.Gid, 10, 32)
+					if err == nil {
+
+						//Check if the recursive flag is set and the path is a directory
+						if chownReq.Recursive {
+
+							err := filepath.WalkDir(target, func(file string, d fs.DirEntry, err error) error {
+								if err == nil {
+									err = os.Chown(file, int(uid), int(gid))
+									if err != nil {
+										return err
+									}
+								} else {
+									return err
+								}
+								return nil
+							})
+							if err != nil {
+								chown.Response.Err = err.Error()
+							}
+
+						} else {
+
+							err = os.Chown(target, int(uid), int(gid))
+							if err != nil {
+								chown.Response.Err = err.Error()
+							}
+						}
+					} else {
+						chown.Response.Err = err.Error()
+					}
+				} else {
+					chown.Response.Err = err.Error()
+				}
+			} else {
+				chown.Response.Err = err.Error()
+			}
+		} else {
+			chown.Response.Err = err.Error()
+		}
+	} else {
+		chown.Response.Err = err.Error()
+	}
```

Comment:
This whole block should be refactored to be easier to read. 90% of this doesn't need to be nested if we change the test condition to `if err != nil { //set error and return}`. We usually try to "return often, return early".